Docs/troubleshooting.md

```diff
@@ -0,0 +1,236 @@
+
+This is an optional configuration that should only be executed if your concentrator needs to use a proxy server to communicate with Azure.
+
+Follow [this guide](/Docs/devguide.md#use-a-proxy-server-to-connect-your-concentrator-to-azure) to:
+
+1. Configure the Docker daemon and the IoT Edge daemon on your device to use a proxy server.
+2. Configure the `edgeAgent` properties in the `config.yaml` file on your device.
+3. Set environment variables for the IoT Edge runtime in the deployment manifest.
+4. Add the `https_proxy` environment variable to the `LoRaWanNetworkSrvModule` in IoT Hub.
+
+## LoRa Device provisioning
+
+A LoRa device is a normal IoT Hub device with some specific device twin tags. You manage it like you would with any other IoT Hub device.
+**To avoid caching issues you should not allow the device to join or send data before it is provisioned in IoT Hub. In case that you did plese follow the ClearCache procedure that you find below.**
+
+### ABP (personalization) and OTAA (over the air) provisioning
+
+- Login in to the Azure portal go to IoT Hub -> IoT devices -> Add
+- Use the DeviceEUI as DeviceID -> Save
+- Click on the newly created device
+- Click on Device Twin menu
+
+- Add the followings desired properties for OTAA:
+
+```json
+"desired": {
+    "AppEUI": "App EUI",
+    "AppKey": "App Key",
+    "GatewayID": "",
+    "SensorDecoder": ""
+  },
+```
+
+Or the followings desired properties for ABP:
+
+**DevAddr must be unique for every device! It is like an ip address for lora.**
+
+```json
+"desired": {
+    "AppSKey": "Device AppSKey",
+    "NwkSKey": "Device NwkSKey",
+    "DevAddr": "Device Addr",
+    "SensorDecoder": "",
+    "GatewayID": ""
+  },
+```
+
+It should look something like this for ABP:
+
+```json
+{
+  "deviceId": "BE7A00000000888F",
+  "etag": "AAAAAAAAAAs=",
+  "deviceEtag": "NzMzMTE3MTAz",
+  "status": "enabled",
+  "statusUpdateTime": "0001-01-01T00:00:00",
+  "connectionState": "Disconnected",
+  "lastActivityTime": "2018-08-06T15:16:32.0658492",
+  "cloudToDeviceMessageCount": 0,
+  "authenticationType": "sas",
+  "x509Thumbprint": {
+    "primaryThumbprint": null,
+    "secondaryThumbprint": null
+  },
+  "version": 324,
+  "tags": {
+  
+  },
+  "properties": {
+    "desired": {
+      "AppSKey": "2B7E151628AED2A6ABF7158809CF4F3C",
+      "NwkSKey": "1B6E151628AED2A6ABF7158809CF4F2C",
+      "DevAddr": "0028B9B9",
+      "SensorDecoder": "",
+      "GatewayID": "",
+      "$metadata": {
+        "$lastUpdated": "2018-03-28T06:12:46.1007943Z"
+      },
+      "$version": 1
+    },
+    "reported": {
+      "$metadata": {
+        "$lastUpdated": "2018-08-06T15:16:32.2689851Z"
+      },
+      "$version": 313
+    }
+  }
+}
+```
+
+- Click Save
+- Turn on the device and you are ready to go
+
+### Optional device properties
+
+Customization's to lora devices are set by creating specific twin desired properties on the device. The following customization's are available:
```

Comment:
Customization's --> remove the appostrophe. This mistake is 2x in the text. It should be "customizations"
lora --> Conistancy --> change to LoRa